public/scripts/backupSheets.js

```diff
@@ -49,39 +43,57 @@ function backupSheets() {
       sheet_id: "1GwP7Ly6iaqgcms0T80QGCNW4y2gJ7tzVND2CktFqnXM",
       backup_folder_id: "1gaKQLiUxeIHGJl3it-sDpAtjGEAhT-wg"
     }];
-  
+
+const SHEETS_FOLDER = "1scY72enARHyBS5nKow6qeT4TLH9feL09";
+
+/*
+  Create a backup of each country's sheet in a country-specific
+  Google Drive folder.
+*/
+function backupSheets() {
+    
   const formattedDate = Utilities.formatDate(new Date(), "PST", "yyyy-MM-dd' 'HH:mm:ss");
   
-  const sheetsFolder = DriveApp.getFolderById("1scY72enARHyBS5nKow6qeT4TLH9feL09");
+  const sheetsFolder = DriveApp.getFolderById(SHEETS_FOLDER);
   
-  for (let i = 0; i < sheets.length; i++) {
-    const name = "Backup of " + sheets[i].name + " " + formattedDate;
-  
-    const file = DriveApp.getFileById(sheets[i].sheet_id);
-  
-    const destination = DriveApp.getFolderById(sheets[i].backup_folder_id);
-  
-    // Makes a copy of file with name "name" in the "destination" folder.
-    file.makeCopy(name, destination);
+  for (let i = 0; i < SHEETS.length; i++) {
+    const name = "Backup of " + SHEETS[i].name + " " + formattedDate;
+    const destination = DriveApp.getFolderById(SHEETS[i].backup_folder_id);
+    const source = SpreadsheetApp.openById(SHEETS[i].sheet_id);
+    
+    copySheet(source, destination, name);
   }
-  
-  // HACKHACKHACK
-  // By default, any time you make a copy of a Google Sheet that is associated
-  // with a form, a copy of the associated form is made.  This function deletes
-  // those copies.
-  cleanupFormCopies(sheetsFolder);
 }
 
-function cleanupFormCopies(folder) {
-  const filesIterator = folder.getFiles();
+/*
+  If we do a simple file copy, we make a copy of the spreadsheet
+  and also the associated form(s) and script files, which is bad.
+  To work around this, we create a new spreadsheet and copy in the tabs
+  (sheets) from the source sheet.
```

Comment:
O.O

Whoa. how in the world did you even think about  doing this. 👏

Reply:
Stack overflow, of course.  Should I cite my source in a comment here?

Reply:
Credit added.